views/js/editor.js

```diff
@@ -0,0 +1,154 @@
+     * 
+     * @param {jQuery} $container
+     * @param {object} options
+     * @param {boolean} [options.readonly] - not editable
+     * @param {boolean} [options.hidden] - initially hidden or not
+     * @param {number|string} [options.top] - position top absolute to the parent
+     * @param {number|string} [options.left] - position left absolute to the parent
+     * @param {number|string} [options.width] - width of the editor
+     * @param {number|string} [options.height] - height of the editor
+     * @param {number} [options.zIndex] - z-index of the editor
+     * @fires change.xml-editor
+     * @returns {object}
+     */
+    function init($container, options){
+
+        var $editor = $('<div>', {'class' : 'tao-ace-editor'}).appendTo($container);
+        var editor = ace.edit($editor[0]);
+
+        options = _.defaults(options || {}, _defaults);
+        
+        editor.$blockScrolling = Infinity;//add this fix as suggested by ace to prevent message in console
+        editor.setTheme("ace/theme/monokai");
+        editor.getSession().setMode("ace/mode/xml");
+        editor.setReadOnly(options.readonly);
+        editor.setShowPrintMargin(false);
+        editor.on('input', _.throttle(function(){
+            $container.trigger('change'+_ns, [getValue()]);
+        }, 600));
+
+        //set editor style options
+        $container
+            .addClass('tao-xml-editor')
+            .css({
+                top : options.top,
+                left : options.left,
+                width : options.width,
+                height : options.height,
+                zIndex : options.zIndex
+            });
+        
+        if(options.hidden){
+            $container.hide();
+        }
+        
+        /**
+         * Set the editor content
+         * @param {string} xml
+         */
+        function setValue(xml){
+            xml = formatXml(xml);
+            editor.getSession().setValue(xml);
+        }
+        
+        /**
+         * Get the editor content
+         *  
+         * @returns {string}
+         */
+        function getValue(){
+            return compressXml(editor.getValue());
+        }
+        
+        /**
+         * Destroy the editor
+         */
+        function destroy(){
+            
+            editor.destroy();
+            $container
+                .off(_ns)
+                .removeClass('tao-xml-editor')
+                .removeAttr('style')
+                .children('.tao-ace-editor').remove();
+        }
+        
+        return {
+            setValue : setValue,
+            getValue : getValue,
+            destroy : destroy,
+            show : function show(){
+                $container.show().trigger('show'+_ns);
+                setValue(getValue());//ensure that the rendering is updated
+            },
+            hide : function hide(){
+                $container.hide().trigger('hide'+_ns);
+            }
+        };
+    }
+
+    return {
+        init : init
```

Comment:
It is just the method returns by the editor and is not a hook on his own.
But https://github.com/oat-sa/extension-tao-xmledit/blob/feature/viewer/views/js/hooks/qtiCreatorDebugger/hook.js#L122-L124 and  https://github.com/oat-sa/extension-tao-xmledit/blob/feature/viewer/views/js/hooks/apipCreatorDebugger/hook.js#L71-L73 are hooks so I added the comment as suggested